povm_toolbox/post_processor/median_of_means.py

```diff
@@ -0,0 +1,196 @@
+    """
+
+    def __init__(
+        self,
+        povm_sample: POVMPubResult,
+        dual: BaseDual | None = None,
+        *,
+        num_batches: int | None = None,
+        delta_confidence: float | None = None,
+        seed: int | Generator | None = None,
+    ) -> None:
+        r"""Initialize the median-of-means post-processor.
+
+        Given ``num_shots=num_batches*batch_size`` samples, we partition the
+        samples into ``num_batches`` batches. We compute the mean of each batch,
+        :math:`\hat{o}_j`, and then output the median of the means, :math:`\hat{o}
+        =\mathrm{median}\{\hat{o}_1, ..., \hat{o}_{\mathrm{num\_batches}}\}`.
+        It can be shown that
+
+        .. math::
+            \lvert \mathrm{Tr}[\mathcal{O} \rho] - \hat{o} \rvert \leq \epsilon
+            \quad \textrm{with probability at least } 1-\delta \, ,
+
+        where :math:`\delta = 2 \exp{(-\mathrm{num\_batches}/2)}` and :math:`\epsilon
+        = \sqrt{\frac{34}{\mathrm{batch\_size}} } \lVert \mathcal{O} - \frac{\mathrm{Tr}
+        [\mathcal{O}]}{2^N} \mathbb{I} \rVert_\textrm{shadow}`.
+
+        Args:
+            povm_sample: a result from a POVM sampler run.
+            dual: the Dual frame that will be used to obtain the decomposition weights of an
+                observable when computing its expectation value. For more details, refer to
+                :meth:`get_decomposition_weights`. When this is ``None``, the canonical Dual frame
+                will be constructed from the POVM stored in the ``povm_sample``'s
+                :attr:`.POVMPubResult.metadata`.
+            num_batches: number of batches, i.e. number of samples means, used in the median-of-means
+                estimator. This value will be overridden if a ``delta_confidence`` argument is supplied.
+            delta_confidence: the confidence parameter :math:`\delta`. It is used to determine the
+                necessary number of batches as :math:`\mathrm{num\_batches} = \lceil 2 \log{(2/\delta)}
+                \rceil`. It will override any ``num_batches`` supplied argument. If both ``num_batches``
+                and ``delta_confidence`` are ``None``, ``delta_confidence`` is set to 0.05.
+            seed: optional seed to fix the :class:`numpy.random.Generator` used to generate the
+                batches. The user can also directly provide a random generator. If ``None``, a
+                random seed will be used.
+
+        Raises:
+            ValueError: If the provided ``dual`` is not a dual frame to the POVM stored in the
+                ``povm_samples``'s :attr:`.POVMPubResult.metadata`.
+            TypeError: If the type of ``seed`` is not valid.
+        """
+        super().__init__(povm_sample=povm_sample, dual=dual)
+
+        if isinstance(delta_confidence, float) or num_batches is None:
+            delta_confidence = delta_confidence or 0.05
+            num_observables = 1
```

Comment:
What is the need for explicitly having this variable and set it to 1?
Should there be a way to set this?

If not, I don't see a reason to keep it.

Reply:
It was looking a bit too much into the future.
If we ever provide an interface where the user can request the expectation values of multiple observables in one call, it would be useful to have this variable but we can remove it for now.